tas-client/src/tas-client/FeatureProvider/TasApiFeatureProvider.ts

```diff
@@ -38,7 +38,48 @@ export class TasApiFeatureProvider extends FilteredFeatureProvider {
         }
 
         //axios webservice call.
-        let response: AxiosResponse<TASFeatureData> = await this.httpClient.get({ headers: headers });
+        let response: AxiosResponse<TASFeatureData> | undefined;
+
+        
+        try {
+            /**
+             * As mentioned in the axios docs:
+             * Axios Promise will handle handle the catch
+             * section of promises whenever a request is not succesful.
+             * https://axios-http.com/docs/handling_errors
+             */
+            response = await this.httpClient.get({ headers: headers });
+        } catch(error) {
+            const axiosError = error as AxiosError;
+            const properties: Map<string, string> = new Map();
+            if (axiosError.response) {
+                // The request was made and the server responded with a status code
+                // that falls out of the range of 2xx
+                properties.set("RESPONSE_DATA", axiosError.response.data);
+                properties.set("RESPONSE_STATUS", `${axiosError.response.status}`);
+                properties.set("RESPONSE_HEADERS", `${axiosError.response.headers}`);
+            } else if (axiosError.request) {
+                // The request was made but no response was received
+                // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
+                // http.ClientRequest in node.js
+                properties.set("ERROR_REQUEST", JSON.stringify(axiosError.request));
```

Comment:
These we also aren't in control of what the messages are. Maybe for a first iteration it's just enough to know if it was a response, request, or other error and we have a property that says which?

Reply:
sounds good, I'll create a property just to point out the type of error .